### ISSUE TITLE
Do you get good prediction when using Dice Score?

### ISSUE THREAD

**Issue by WMeixiang** [ouank/PyTorch_UNet#1]
Hi, I am Interested in Attention-Net.
1）Have you run your code on  the NIH Pancreas-CT dataset before ？ How about your result of prediction.
2）How do you handle your data?  Just slice?
3）what's your data shape you feed into network? Have you do crop operation?

**Comment by ouank** [comment]
Hi, 
1) Yes I ran it on the NIH Pancreas CT dataset for Pancreas Segmentation. As you can see in my code I used 2D convolutions, instead of 3D (Due to limitations of GPU). Metrics were compareable to the paper (A little bit worse which could be because I used 2D instead of 3D). If you havent read the paper Attention Unet, that I have linked, I would strongly recommend it. 
2) Yes I sliced the Axial view into 2D slices and did Data Augmentation.
3) As I just used the Axial view, the shape for all data was always the same. Just the amount of slices varies from patient to patient.

**Comment by WMeixiang** [comment]
Thank you very much for your reply.
 I have read the paper Attention U-net, whose training data is not NIH  Pancreas CT dataset in their code. I really want to reproduce your code.
I use unet_nonlocal_2D model, I also used the Axial view while I didn't do Data Augmentation. Because I just sliced, the single data shape is [512,512], I didn't do any crop or resize. My loss is SoftDiceLoss. 
But there is something wrong when I train, the loss is very high (actually: 0.996) and basically unchanged during the training process. I try to adjust the learning rate, but it  didn't make it well.
I suppose if I need do some data processing?  Could you give me some advice?  Thank you very much.

**Comment by ouank** [comment]
Yes actually they did train it on another dataset first, but they also report the results when training on NIH dataset from scratch.
When the Loss is not changing at all, it can have various reasons. And there are a lot of blogs you can find and follow on the internet, which tells you where you can have a look at. 
Just as some examples what you can check for: 
- What kind of learning rates did you try? Did you think about sampling different learning rates from a grid or a uniform distribution? 
- Check your regularization. Maybe try to overfit at first to check if your network is learning at all? 
- Check the data you feed into the network and how it fits to your loss function. For example are the gray values between 0-255 or normalized to 0-1? 
- At what time did you report the loss of 0.996? 
- Check the Loss function again if you wrote it yourself.
Hope that helps.

**Comment by WMeixiang** [comment]
Thank you very much for your suggestion, I will try again.




Best wishes,

Meixiang Huang

-----原始邮件-----
发件人:ouank <notifications@github.com>
发送时间:2019-11-10 21:04:37 (星期日)
收件人: ouank/PyTorch_UNet <PyTorch_UNet@noreply.github.com>
抄送: WMeixiang <11735032@zju.edu.cn>, Author <author@noreply.github.com>
主题: Re: [ouank/PyTorch_UNet] Do you get good prediction when using Dice Score? (#1)



Yes actually they did train it on another dataset first, but they also report the results when training on NIH dataset from scratch.
When the Loss is not changing at all, it can have various reasons. And there are a lot of blogs you can find and follow on the internet, which tells you where you can have a look at.
Just as some examples what you can check for:

What kind of learning rates did you try? Did you think about sampling different learning rates from a grid or a uniform distribution?
Check your regularization. Maybe try to overfit at first to check if your network is learning at all?
Check the data you feed into the network and how it fits to your loss function. For example are the gray values between 0-255 or normalized to 0-1?
At what time did you report the loss of 0.996?
Check the Loss function again if you wrote it yourself.
Hope that helps.

**Comment by WMeixiang** [comment]
Hi, I obtain a model through training, the training loss and Dice is normal. Then I load model parameters for testing on a small part of NIHDataset, but I got a very much bad result, Could you write a test.py for reference? Thank you
